x-pack/plugins/infra/public/pages/link_to/redirect_to_node_detail.tsx

```diff
@@ -7,51 +7,58 @@
 
 import React from 'react';
 import { Redirect, useLocation, useRouteMatch } from 'react-router-dom';
+import rison from '@kbn/rison';
+import { replaceStateKeyInQueryString } from '../../../common/url_state_storage_service';
 import { replaceMetricTimeInQueryString } from '../metrics/metric_detail/hooks/use_metrics_time';
-import {
-  getFromFromLocation,
-  getToFromLocation,
-  getNodeNameFromLocation,
-  getStateFromLocation,
-} from './query_params';
 import { InventoryItemType } from '../../../common/inventory_models/types';
-import { RouteState } from '../../components/asset_details/types';
+import { AssetDetailsUrlState } from '../../components/asset_details/types';
+import { ASSET_DETAILS_URL_STATE_KEY } from '../../components/asset_details/constants';
+
+export const REDIRECT_NODE_DETAILS_FROM_KEY = 'from';
+export const REDIRECT_NODE_DETAILS_TO_KEY = 'to';
+export const REDIRECT_ASSET_DETAILS_KEY = 'assetDetails';
+
+const getHostDetailSearch = (queryParams: URLSearchParams) => {
+  const from = queryParams.get(REDIRECT_NODE_DETAILS_FROM_KEY);
+  const to = queryParams.get(REDIRECT_NODE_DETAILS_TO_KEY);
+  const assetDetailsParam = queryParams.get(REDIRECT_ASSET_DETAILS_KEY);
+
+  return replaceStateKeyInQueryString(ASSET_DETAILS_URL_STATE_KEY, {
+    ...(assetDetailsParam ? (rison.decode(assetDetailsParam) as AssetDetailsUrlState) : undefined),
+    dateRange: {
+      from: from ? new Date(parseFloat(from)).toISOString() : undefined,
+      to: to ? new Date(parseFloat(to)).toISOString() : undefined,
+    },
+  } as AssetDetailsUrlState)('');
+};
+
+const getNodeDetailSearch = (queryParams: URLSearchParams) => {
+  const from = queryParams.get(REDIRECT_NODE_DETAILS_FROM_KEY);
+  const to = queryParams.get(REDIRECT_NODE_DETAILS_TO_KEY);
+
+  return replaceMetricTimeInQueryString(
+    from ? parseFloat(from) : NaN,
```

Comment:
Q: should we have a fallback to default in case if the `from` is not present instead 🤔

Reply:
The [`replaceMetricTimeInQuery` ](https://github.com/elastic/kibana/blob/main/x-pack/plugins/infra/public/pages/metrics/metric_detail/hooks/use_metrics_time.ts#L107) function handles it. This new way works exactly the same way it worked before, passing `NaN` when `from` or `to` are not present.

Reply:
Got it, thanks for explaining 👍